### PR TITLE
Feat/contractor crm

### PR DESCRIPTION
**Pull request by KornelijusGlinskas**
*** DONE ***
- [x] Add icons font
- [x] Create separate twig template for admin
- [x] Refactor SCSS for admin CSS
- [x] Add new entry for admin JS and CSS
- [x] Create desktop sidebar nav
- [x] Create mobile sidebar nav
- [x] Style desktop view
- [x] Style mobile view
- [x] Connect React
- [x] Setup Router

### CODE REVIEW

File: assets/js/modals/admin/edit-row.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+// const editBtn = document.querySelectorAll('.js-edit');
+// const editWindow = document.querySelector('.js-edit-window');
+
+// export default function initEditRow() {
+//   editBtn.forEach((e) =>
+//     e.addEventListener('click', () => editWindow.classList.toggle('-open'))
+//   );
+// }
```
Comment (tadas-nfq) [inline assets/js/modals/admin/edit-row.js]: Commented code should be removed or changed with working code.

File: assets/js/modals/admin/search.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+// const search = document.querySelector('.js-search');
+// const searchInput = document.querySelector('.js-search-input');
+
+// export default function initSearch() {
+//   search.addEventListener('click', () => {
+//     searchInput.focus();
+//   });
+// }
```
Comment (tadas-nfq) [inline assets/js/modals/admin/search.js]: Commented code should be removed or changed with working code.

File: assets/js/react.jsx [hunk 3]
```diff
@@ -1,7 +1,7 @@
...
 import { render } from 'react-dom';
 
-render(
-  <h1>Hello, world</h1>,
-  document.getElementById('contractor__reservations'),
-);
+// render(
+//   <h1>Hello, world</h1>,
+//   document.getElementById('contractor__reservations'),
+// );
```
Comment (tadas-nfq) [inline assets/js/react.jsx]: Commented code should be removed or changed with working code.

File: assets/js/components/admin/Panel.jsx [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+            aria-label="close menu"
+            tabIndex="0"
+            className="overlay js-overlay"
+            onClick={() => toggleNav(!isNavOpen)}
+            onKeyPress={() => toggleNav(!isNavOpen)}
+          />
+          <PanelHeader isOpen={isNavOpen} toggleNav={toggleNav} />
+          <Switch>
+            <Route path="/contractor/reservations" component={Reservations} />
+            <Route path="/contractor/settings" component={Settings} />
```
Comment (tadas-nfq) [inline assets/js/components/admin/Panel.jsx]: Routes should be generated dynamically. You can use this bundle to generate routes from backend
https://symfony.com/doc/master/bundles/FOSJsRoutingBundle/usage.html

File: assets/js/components/admin/Sidenav.jsx [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+        onKeyPress={() => toggleNav(!isOpen)}
+      />
+      <img
+        src={logoPath}
+        alt="nnizer typo logotype"
+        className="sidenav__logo"
+      />
+      <ul className="sidenav__links">
+        <li className="sidenav__link -active">
+          <NavLink to="/contractor/reservations" onClick={() => toggleNav(false)}>
```
Comment (tadas-nfq) [inline assets/js/components/admin/Sidenav.jsx]: Routes should be generated dynamically. You can use this bundle to generate routes from backend
https://symfony.com/doc/master/bundles/FOSJsRoutingBundle/usage.html

File: assets/js/components/admin/Panel.jsx (hunk 4 above)
Comment (KornelijusGlinskas) [reply assets/js/components/admin/Panel.jsx]: Now the path exists only in react memory. It doesn't change the URL directly. Not sure if it is good practice, however it works now.

**Comment by dominykasmurauskas** [comment]
Some suggestions:

Add a link on logo to homepage:
image

On homepage, add a button to access service provider dashboard, because now it's not possible to enter it:
image

**Comment by dominykasmurauskas** [comment]
Before merging, language and email arguments should be binded. Let me know before merging to master so that I can fix it.

**Comment by KornelijusGlinskas** [comment]
> Some suggestions:
> 
> Add a link on logo to homepage:
> image
> 
> On homepage, add a button to access service provider dashboard, because now it's not possible to enter it:
> image
> 
> Log out button should redirect to /logout, now it's redirecting to homepage.
> 
> Also, I believe contractor should automatically have "Reservations" tab active after log in instead of the blank page.

All things done except button to dashboard. I think it is better to do it in the next PR because this is getting way too big.

**Pull request merged**